src/v2/coins/eth.js

```diff
@@ -246,6 +246,54 @@ class Eth extends BaseCoin {
     }).call(this).asCallback(callback);
   }
 
+  /**
+   * Helper function for signTransaction for the rare case that SDK is doing the second signature
+   * params:
+   * - txPrebuild
+   * - prv
+   * Note: we are expecting this to be called from the offline vault
+   */
+  signFinal(params) {
+    const txPrebuild = params.txPrebuild;
+
+    if (!_.isNumber(params.signingKeyNonce)) {
+      throw new Error('must have signingKeyNonce as a parameter, and it must be a number');
+    }
+    if (_.isUndefined(params.walletContractAddress)) {
+      throw new Error('params must include walletContractAddress, but got undefined');
+    }
+
+    const signingNode = prova.HDNode.fromBase58(params.prv);
+    const signingKey = signingNode.getKey().getPrivateKeyBuffer();
+
+    const txInfo = {
+      recipient: txPrebuild.recipients[0],
+      expireTime: txPrebuild.halfSigned.expireTime,
+      contractSequenceId: txPrebuild.halfSigned.contractSequenceId,
+      signature: txPrebuild.halfSigned.signature
+    }
+
+    const sendMethodArgs = this.getSendMethodArgs(txInfo);
+    const methodSignature = ethAbi.methodID('sendMultiSig', _.map(sendMethodArgs, 'type'));
+    const encodedArgs = ethAbi.rawEncode(_.map(sendMethodArgs, 'type'), _.map(sendMethodArgs, 'value'));
+    const sendData = Buffer.concat([methodSignature, encodedArgs]);
+
+    const ethTxParams = {
+      to: params.walletContractAddress,
+      nonce: params.signingKeyNonce,
+      value: 0,
+      gasPrice: new ethUtil.BN(txPrebuild.gasPrice),
+      gasLimit: new ethUtil.BN(txPrebuild.gasLimit),
+      data: sendData,
+      spendAmount: params.recipients[0].amount
+    }
+
+    const ethTx = new EthTx(ethTxParams);
+    ethTx.sign(signingKey);
+    const txHex = ethTx.serialize().toString('hex');
+    return { txHex };
```

Comment:
It's a little cleaner if you do `return { txHex: ethTx.serialize().toString('hex') };` instead.